Packs/Jira/Integrations/JiraV3/JiraV3_test.py

```diff
@@ -0,0 +1,1954 @@
+        Then
+            - Validate that the correct data is extracted and returned
+        """
+        from JiraV3 import add_extracted_data_to_incident
+        issue = util_load_json('test_data/get_issue_test/raw_response.json')
+        add_extracted_data_to_incident(issue=issue)
+        expected_extracted_issue_data = {'extractedSubtasks': [
+            {'id': '21525', 'key': 'COMPANYSA-63'}, {'id': '21538', 'key': 'COMPANYSA-70'}],
+            'extractedCreator': 'Example User(admin@demistodev.com)', 'extractedComponents': [
+                'Almost-Done', 'dummy-comp', 'Integration', 'New-Component']}
+        assert expected_extracted_issue_data.items() <= issue.items()
+
+    @pytest.mark.parametrize('issue_field_priority, severity', [
+        ({'name': 'Highest'}, 4),
+        ({'name': 'High'}, 3),
+        ({'name': 'Medium'}, 2),
+        ({'name': 'Low'}, 1),
+        ({'name': 'Lowest'}, 1),
+        ({'name': 'Extreme'}, 0),
+    ])
+    def test_get_jira_issue_severity(self, issue_field_priority, severity):
+        """
+        Given:
+            - The priority field of an issue
+        When
+            - Determining the severity of the incident
+        Then
+            - Validate that the priority of the issue is mapped to the correct severity
+        """
+        from JiraV3 import get_jira_issue_severity
+        assert severity == get_jira_issue_severity(issue_field_priority)
+
+    def test_parse_custom_fields(self):
+        """
+        Given:
+            - An issue response
+        When
+            - Parsing the custom fields in the response to a more human readable form
+        Then
+            - Validate that the data of the custom fields get parsed and show the correct data
+        """
+        from JiraV3 import parse_custom_fields
+        issue = util_load_json('test_data/get_issue_test/raw_response.json')
+        expected_parsed_custom_fields = util_load_json('test_data/parsed_issue_custom_fields.json')
+        parse_custom_fields(issue=issue, issue_fields_id_to_name_mapping=issue.get('names', {}) or {})
+        assert expected_parsed_custom_fields == issue
+
+    def test_set_last_run_when_first_time_running_fetch(self, mocker):
+        """
+        Given:
+            - Arguments to use when calling the fetch incidents mechanism
+        When
+            - Calling the fetch incidents mechanism for the first time (last_run is empty)
+        Then
+            - Validate that the last run object gets saved with the correct data
+        """
+        from JiraV3 import (fetch_incidents, DEFAULT_FETCH_LIMIT)
+        client = jira_base_client_mock()
+        mocker.patch('JiraV3.demisto.getLastRun', return_value={})
+        mocker.patch('JiraV3.create_incident_from_issue', return_value={})
+        set_last_run_mocker = mocker.patch('JiraV3.demisto.setLastRun', side_effect=demisto.setLastRun)
+        query_raw_response = util_load_json('test_data/issue_query_test/raw_response.json')
```

Comment:
The first query_raw_response var is not used (relevant to this tests and some other).